lib/timeutils/zoneinfo.c

```diff
@@ -462,15 +462,15 @@ zone_info_get_offset(ZoneInfo *self, gint64 timestamp)
 
   if (self->last_transitions_index != -1 &&
       self->last_transitions_index < (self->timecnt - 1) &&
-      self->transitions[self->last_transitions_index].time < timestamp &&
+      self->transitions[self->last_transitions_index].time <= timestamp &&
```

Comment:
Could you explain this patch a bit more?

I am not familiar with this transition table feature. Can you explain what is it used for?

Reply:
This loop is processing the daylight saving periods, e.g. when dst is enabled/disabled. The current time and the table is measured in UTC

Reply:
Also, the table contains periods that are back to back, e.g. no seconds in between. And the off by one causes the right entry to be skipped.

You can check the tables using the zdump tool.